app/app.go

```diff
@@ -626,6 +626,12 @@ func (app *TerraApp) BeginBlocker(ctx sdk.Context, req abci.RequestBeginBlock) a
 		}
 	}
 
+	// check if version map in upgrade keeper is nil
+	vm := app.UpgradeKeeper.GetModuleVersionMap(ctx)
```

Comment:
Don't we need to wrap this logic in a "blockheight guard"? As such GetModuleVersionMap calls the underlying KVStore @ https://github.com/cosmos/cosmos-sdk/blob/f2b6013cee0fd4cd0d3751a173f1344cdd718e67/x/upgrade/keeper/keeper.go#L155 so I suppose this will generate another transaction that we dont want triggering before a certain target block height or I suppose it would alter the TX  signatures when nodes use the 1.0.5 client to sync from columbus-5 genesis? :)

Reply:
It is `reading` values from the store. not making a tx

Reply:
You are partially right. I think there should be a height guard for `SetModuleVersionMap` which changes the state of the chain.

Reply:
ok will have to wrap in block height guards then...